### DISCUSSION TITLE
Show light state attributes when bulbs are off

### DISCUSSION THREAD

**Discussion by emichael** [home-assistant/architecture#535] [Entity Models]
Related to https://github.com/home-assistant/core/pull/46605.

Currently, light entities hide their state attributes from HA when they are off. This might be simpler in most cases, but the data is sometimes useful. It would be better to expose all of the data to the user for access in HA automations.

Initially, I [submitted a PR which changed the Light entity model entirely](https://github.com/home-assistant/core/pull/46605/commits/e39930ef015b1327b79073bf870fb7e5af5fca3b), exposing the state attributes even when the light is off. This breaks tests for quite a few integrations.

I also proposed [modifying the Light entity in an unobtrusive way to allow specific integrations to expose state attributes when bulbs are in the off state](https://github.com/home-assistant/core/pull/46605/files). (In my case, I wanted to change the LIFX Light entity).

The second approach is certainly a less breaking change, as lights could be modified on an integration-by-integration basis.

**Comment by mguaylam** [comment]
I think the best representation of this is with Flux. When lights are turned on from a full cycle, they light up at the previous set temperature (very yellow in the morning) and then, switching to the expected cold white.

Showing the previous brightness and colour temperature and or colour seems to me logical as it is the values at which the device will light to when turned back on (at least for ZHA)

**Reply by icemanch** [reply]
The other case would be if you dimmed the lights at night and then turned them off. If you turn them on again in the morning they would be at that dimmed state. Depending on how the integration handled it of course.  If the state attributes where shown then you could set it up so that they return to full brightness when turned back on or leave it as is. 

I guess this could be added to the integration itself but, it would be better for the user to be able to see and adjust the settings from the UI.

**Comment by thecode** [comment]
If we want to show the attributes when the lights are off  it might be also needed to additional flag that tells the entity if the device store the last attributes when off. Although I believe most device do, I assume some of the devices always turn on to a default state and not to the last state that they turned off at, In addition some devices might not expose all the attributes when off, so the first HA restart they will be missing.
The flag itself can set as as a `supported_features`. If it is designed like this than unless an integration add this flag the current implemented behavior will not change.

**Comment by AndreKR** [comment]
Couldn't this simply be added as an additional attribute? Like `previous_state`? That way it can be accessed but isn't a breaking change.

**Reply by parautenbach** [reply]
What if a light has a fade setting? Not all attribute values when off will be reflecting the history. Some are current and will impact the future.
